### ISSUE TITLE
Use PHPStan for static analysis

### ISSUE THREAD

**Issue by szepeviktor** [getkirby/kirby#1990]
Hello!

Many PHPDoc blocks have wanna-be absolute classes `Kirby\Cms\File`
they should really be `\Kirby\Cms\File`

```bash
grep -rFl '@return Kirby' config/ src/ | xargs -t -L1 sed -e 's#@return Kirby#@return \\Kirby#' -i

grep -rFl '@param Kirby' config/ src/ | xargs -t -L1 sed -e 's#@param Kirby#@param \\Kirby#' -i

grep -rFl '@var Kirby' config/ src/ | xargs -t -L1 sed -e 's#@var Kirby#@var \\Kirby#' -i

grep -rFl '|Kirby\' config/ src/ | xargs -t -L1 sed -e 's#|Kirby\\#|\\Kirby\\#' -i
```

After that you may gain the huge benefit of @phpstan as this tool works based on types -  although it takes couple of minutes to fix.

Its simplest form is `phpstan analyze -l 0 src/`

Levels go up to 7.

An example: Cms/PageBlueprint.php:79 Array has 2 duplicate keys with value 0 (0, '0').

Another two: Cms/ModelWithContent.php:399 Access to undefined constant Kirby\Cms\ModelWithContent::CLASS_ALIAS.

Cms/FileVersion.php:92 Kirby\Cms\FileVersion::toArray() calls parent::toArray() but Kirby\Cms\FileVersion does not extend any class.

**Comment by sanderdlm** [comment]
Has this issue been tackled? Running PHPStan myself I get a few errors but nothing relating to ```Kirby\Cms\File```.

**Comment by szepeviktor** [comment]
A moment please, I run it again...

**Comment by szepeviktor** [comment]
@distantnative is a tricky guy! Meanwhile he fixed everything silently :)

e.g. https://github.com/getkirby/kirby/commit/3e63f931513d4d47ca245fa6e830f35fc3486720

**Comment by szepeviktor** [comment]
I wish you a successful journey from Level 0 up to Level "max"!

**Comment by sanderdlm** [comment]
I guess you could add PHPStan to the travis CI flow at this point?

Most of the errors are related to return type and docblocks, but there's also ```PermissionException``` use in ```Cms/Auth.php``` where 2 parameters are passed (a message string and a HTTP code), but that exception extends ```Exception``` which is defined to take one array as parameters, with the message/httpcode inside.

Can/should these errors be fixed? And in separate issues or this one?

**Comment by szepeviktor** [comment]
Return type and docblocks errors are the foundation. On higher Levels you will receive direction from @phpstan to make your project more robust. I suggest at least Level 4.

**Comment by szepeviktor** [comment]
I think this issue is about reaching error-free state on Level 4 and including phpstan in Travis.

**Comment by bastianallgeier** [comment]
We discussed this and decided to move forward step by step. As a first step we've ramped up our PHPCS config, which already solves a few things. Including the absolute class names. 

But this process is not going to happen over night. We have to move carefully to not produce more issues than we actually solve, just by making the machine happy. I think we will keep this issue open for as long as we are working on it and see it as a work-in-progress reminder.
